addons/account_edi/models/account_move.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models
+
+
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+
+    @api.returns('mail.message', lambda value: value.id)
+    def message_post(self, **kwargs):
+        # OVERRIDE
+        # /!\ 'default_res_id' in self._context is used to not process attachment when using a form view.
+        res = super(AccountMove, self).message_post(**kwargs)
+
+        if len(self) != 1 or self.env.context.get('no_new_invoice') or not self.is_invoice(include_receipts=True):
+            return res
+
+        attachments = self.env['ir.attachment'].browse(kwargs.get('attachment_ids', []))
+        odoobot = self.env.ref('base.partner_root')
+        if attachments and self.state != 'draft':
+            self.message_post(body='The invoice is not a draft, it was not be updated from the attachment.',
+                              message_type='comment',
+                              subtype_xmlid='mail.mt_note',
+                              author_id=odoobot.id)
+            return res
+        if attachments and self.invoice_line_ids:
+            self.message_post(body='The invoice already contains lines, it was not be updated from the attachment.',
+                              message_type='comment',
+                              subtype_xmlid='mail.mt_note',
+                              author_id=odoobot.id)
+            return res
+
+        for attachment in attachments:
+            invoice = self.env['account.edi'].search([])._update_invoice_from_attachment(attachment, self)
+            if invoice:
+                if 'extract_state' in invoice:
+                    invoice.write({'extract_state': 'done'})
```

Comment:
should be done by `_update_invoice_from_attachment` (or `_import_from_attachment`)